src/core/transaction.c

```diff
@@ -1102,13 +1124,13 @@ int transaction_add_job_and_dependencies(
         }
 
         return 0;
```

Comment:
hmm, i think it would be nicer if we'd simply collect all PropagatesStopTo= units in a new set during the entire transaction, and then once we are done with compiling the transaction we go through that set and enqueue a stop job for that unit, but gracefully, i.e. that if there's already a job for the same unit applied in the transaction we'll simply skip it.

Reply:
this would require splitting the atom UNIT_ATOM_PROPAGATE_STOP I guess. i.e. add  UNIT_ATOM_PROPAGATE_STOP_GRACEFUL as a separate atom, that is used by UNIT_PROPAGATES_STOP_TO, and results in this behaviour.

This way we can clearly say that the propagates-stop-to thing is a "second-class citizen", i.e. that it only has effect on a transaction if there's nothing else already having an effect on the same unit in the transaction.

does that make sense?

Reply:
> i.e. that if there's already a job for the same unit applied in the transaction we'll simply skip it.

But IIUC that defeats the purpose of the original commit, which allows `restart` jobs to be propagated to other units through `PropagatesStopTo=`? If the anchor unit A has `Requires + PropagatesStopTo` on another unit B, a start job will be enqueued for B, and with your suggestion we just ignore B when processing `PropagatesStopTo=`.